tests/datasets/entities/test_entity_key.py

```diff
@@ -0,0 +1,14 @@
+import pytest
+
+from snuba.datasets.entities.entity_key import REGISTERED_ENTITY_KEYS, EntityKey
+from snuba.datasets.entities.factory import reset_entity_factory
+
+
+def test_entity_key() -> None:
+    reset_entity_factory()
+    with pytest.raises(AttributeError):
+        EntityKey.NON_EXISTENT_ENTITY
+    assert REGISTERED_ENTITY_KEYS == {
+        "GENERIC_METRICS_DISTRIBUTIONS": "generic_metrics_distributions",
+        "GENERIC_METRICS_SETS": "generic_metrics_sets",
+    }
```

Comment:
I don't love that this is going to require changing for every entity added to `snuba/datasets/configuration/`. Not sure of a better way to test this behavior but I think that we should not assume the entity list will be so static

Reply:
We will definitely need to figure this out, because it also implies that if a product person wants to add a new entity they will also have to edit this test. The point of the configuration is that changing them doesn't require Python changes.